website/source/docs/internals/anti-entropy.html.markdown

```diff
@@ -0,0 +1,131 @@
+health.
+
+The catalog is maintained only by server nodes. This is because the catalog is
+replicated via the [Raft log](/docs/internals/consensus.html) to provide a
+consolidated and consistent view of the cluster.
+
+<a name="anti-entropy"></a>
+### Anti-Entropy
+
+Now that we have covered the functions of the agent and the catalog, we
+need a mechanism to perform synchronization between the two. In Consul, this
+method is known as anti-entropy.
+
+Anti-entropy is a syncronization of the local agent state and the catalog. For
+example, when a user registers a new service or check with the agent, the agent
+in turn notifies the catalog that this new check exists. Similarly, when a check
+is deleted from the agent, it is consequently removed from the catalog as well.
+
+Anti-entropy is also used to update availability information. As agents run
+their health checks, their status may change, in which case their new status
+is synced to the catalog. Using this information, the catalog can respond
+intelligently to queries about its nodes and services based on their
+availability.
+
+During this synchronization, the catalog is also checked for correctness. If
+any services or checks exist in the catalog that the agent is not aware of, they
+will be automatically removed to make the catalog reflect the proper set of
+services and health information for that agent.
+
+### Periodic Synchronization
+
+In addition to running when changes to the agent occur, anti-entropy is also a
+long-running process which periodically wakes up to sync service and check
+status to the catalog. This ensures that the catalog closely matches the agent's
+true state.
+
+The amount of time between periodic anti-entropy runs will vary based on cluster
+size to avoid saturation. The table below describes the periodic sync times and
+how they change as the Consul cluster grows.
+
+<table class="table table-bordered table-striped">
+  <tr>
+    <th>Cluster Size</th>
+    <th>Periodic Sync Interval</th>
+  </tr>
+  <tr>
+    <td>1 - 128</td>
+    <td>15 seconds</td>
+  </tr>
+  <tr>
+    <td>129 - 256</td>
+    <td>30 seconds</td>
+  </tr>
+  <tr>
+    <td>257 - 512</td>
+    <td>45 seconds</td>
+  </tr>
+  <tr>
+    <td>513 - 1024</td>
+    <td>1 minute</td>
+  </tr>
+  <tr>
+    <td>...</td>
+    <td>...</td>
+  </tr>
+</table>
+
+The intervals above are approximate. Each Consul agent will choose a randomly
+staggered start time within the interval window to avoid a thundering herd.
+
+### Best-effort sync
+
+Anti-entropy can fail in a number of cases, including misconfiguration of the
+agent or its operating environment, I/O problems (full disk, filesystem
+permission, etc.), networking problems (agent cannot communicate with server),
+among others. Because of this, the agent attempts to sync in best-effort
+fashion.
+
+If an error is encountered during an anti-entropy run, the error is logged and
+the agent continues to run. It is possible that it will be reconciled on
```

Comment:
Replace " It is possible that it will be reconciled on a later attempt by the periodic anti-entropy sync" with "The anti-entropy mechanism is run periodically to automatically recover from these types of transient failures."